C/sort/bucketSort.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * C Program to Sort Array using Bucket Sort
+ */
+#include <stdio.h>
+#include <stdlib.h>
+
+/* Structing a Bucket*/
+struct bucket
```

Comment:
There is no need of a structure, I guess you are trying to store some values with respect to some other values, just like hashmaps, but for that, you can use an array and store the value for I at array[i]. that will reduce the line of codes as well.